baiji/copy.py

```diff
@@ -333,7 +359,7 @@ def upload_multipart(self):
     def upload_direct(self):
         import math
         from baiji.util.with_progressbar import FileTransferProgressbar
-        key = self.dst.create()
+        key = self.dst.create(version_id=self.version)
```

Comment:
I don't think this makes sense -- if you're uploading, it should make a new version, so you should only be able to specify the version for the source of the copy. Versions are immutable, enforced by s3, so if you try to write to a specific version I'd expect this to break... Can you test this?